jetty-servlet/src/main/java/org/eclipse/jetty/servlet/listener/ContainerInitializer.java

```diff
@@ -0,0 +1,166 @@
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.servlet.listener;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+import javax.servlet.ServletContainerInitializer;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+
+/**
+ * Utility Methods for manual execution of {@link javax.servlet.ServletContainerInitializer} when
+ * using Embedded Jetty.
+ */
+public final class ContainerInitializer
+{
+    /**
+     * Utility Method to allow for manual execution of {@link javax.servlet.ServletContainerInitializer} when
+     * using Embedded Jetty.
+     *
+     * <code>
+     * ServletContextHandler context = new ServletContextHandler();
+     * ServletContainerInitializer corpSci = new MyCorporateSCI();
+     * context.addEventListener(ContainerInitializer.asContextListener(corpSci));
+     * </code>
+     *
+     * <p>
+     * The {@link ServletContainerInitializer} will have its {@link ServletContainerInitializer#onStartup(Set, ServletContext)}
+     * method called with the manually configured list of {@code Set<Class<?>> c} set.
+     * In other words, this usage does not perform bytecode or annotation scanning against the classes in
+     * your {@code ServletContextHandler} or {@code WebAppContext}.
+     * </p>
+     *
+     * @param sci the {@link ServletContainerInitializer} to call
+     * @return the {@link ServletContextListener} wrapping the SCI
+     * @see SCIAsContextListener#addClasses(Class[])
+     * @see SCIAsContextListener#addClasses(String...)
+     * @see SCIAsContextListener#setClassLoader(ClassLoader)
+     */
+    public static SCIAsContextListener asContextListener(ServletContainerInitializer sci)
+    {
+        return new SCIAsContextListener(sci);
+    }
+
+    public static class SCIAsContextListener implements ServletContextListener
+    {
+        private final ServletContainerInitializer sci;
+        private ClassLoader classLoader;
+        private Set<String> classNames;
+        private Set<Class<?>> classes = new HashSet<>();
+        private Consumer<ServletContext> initConsumer;
+
+        public SCIAsContextListener(ServletContainerInitializer sci)
+        {
+            this.sci = sci;
+        }
+
+        public SCIAsContextListener setClassLoader(ClassLoader classLoader)
+        {
+            this.classLoader = classLoader;
```

Comment:
is this really necessary? When would a loader different to the contexts loader be used?

Reply:
When the classes you are passing in are String references and you want to use the Classloader of whatever context you want to use.
Or do you feel that the `Thread.currentThread().getContextClassLoader()` is correct at the time `ServletContextListener.contextInitialized(ServletContextEvent sce)` is called is sane, and just use that one?

Reply:
nudge.  (prior comment)

Reply:
Removed custom classloader, relying on context classloader during the call to `ServletContextListener.contextInitialized(ServletContextEvent sce)`